clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -404,6 +411,12 @@ private void maybeTriggerWakeup() {
             throw new WakeupException();
         }
     }
+    
+    private void maybeThrowInterruptException() {
```

Comment:
To clarify a bit on why interrupts and wakeup semantics are a little different: If a thread calls wakeup while the consumer thread is in `close()`, the wakeup can be ignored and the commit can be allowed to finish. If a thread calls interrupt on the consumer thread while it is in `close()`, the interrupted thread becomes unable to progress in the commit, since Selector is nonfunctional if the calling thread is interrupted. 

So we really only have two choices in that case. Either we stop trying to commit (current solution), or we figure out a way to clear the interrupt flag, finish the commit, and set the interrupted state again before returning from `close()`. I prefer the current solution, since interrupts are usually meant to stop threads as soon as possible.